openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -232,7 +232,6 @@ TCN_IMPLEMENT_CALL(jlong, SSLContext, make)(TCN_STDARGS, jint protocol, jint mod
     c->mode     = mode;
     c->ctx      = ctx;
     c->pool     = p;
-    SSL_CTX_set_options(c->ctx, SSL_OP_ALL);
```

Comment:
OP_ALL is implementation specific (not well defined as to what it includes and what it doesn't) and may include things that we don't want. For example in OpenSSL 1.0.2 it enables `SSL_OP_LEGACY_SERVER_CONNECT` which allows for insecure renegotiation ... I have been advised to avoid this https://github.com/openssl/openssl/issues/2551#issuecomment-277543068.

I don't think this is a good default and if someone needs it I would prefer to add support for the specific options and allow them to manually enable these, but not do it by default.